src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -216,125 +221,130 @@ const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
   if (verificationStatus === "VERIFIED_NO_CHANGE")
     return <div data-testid="emptyAddressVerification"></div>
 
-  if (addressOptions.length === 0) return null
+  if (!modalType || addressOptions.length === 0) return null
 
-  if (modalType === ModalType.SUGGESTIONS) {
-    return (
-      <ModalDialog title="Confirm your delivery address" onClose={handleClose}>
-        <Text>
-          To ensure prompt and accurate delivery, we suggest a modified shipping
-          address.
-        </Text>
-        <Spacer y={2} />
-        <RadioGroup
-          defaultValue={selectedAddressKey || ""}
-          onSelect={(selected: AddressOption["key"]) => {
-            setSelectedAddressKey(selected)
-          }}
-        >
-          {addressOptions.map(
-            ({ key, recommended, ...addressOption }) =>
-              addressOption.lines && (
-                <BorderedRadio
-                  key={key}
-                  value={key}
-                  label={
-                    recommended
-                      ? AddressSuggestionRadioButton.recommended
-                      : AddressSuggestionRadioButton.user_address
-                  }
-                >
-                  <Flex flexDirection="column">
-                    {addressOption.lines.map((line: string) => (
-                      <Text variant="xs" key={line}>
-                        {line}
-                      </Text>
-                    ))}
-                  </Flex>
-                </BorderedRadio>
-              )
-          )}
-        </RadioGroup>
-        <Spacer y={4} />
-        <Flex width="100%" justifyContent="space-between">
-          <Button onClick={handleBackToEdit} variant="secondaryBlack" flex={1}>
-            Back to Edit
-          </Button>
-          <Spacer x={1} />
-          <Button
-            disabled={!(selectedAddressKey && selectedAddressKey.length > 0)}
-            onClick={() => {
-              if (selectedAddressKey) {
+  return (
+    <ModalDialog
+      width={["100%", 590]}
+      height={["100vh", "auto"]}
+      m={[0, 2]}
+      title={modalTitles[modalType]}
+      onClose={handleClose}
+    >
```

Comment:
These are the relevant changes - I moved the check for `modalType` inside of this wrapping modal component because changing the styles in two places was getting annoying and causing me to make tedious mistakes.